5_distance_analysis/nbconverted/1_centroid_distance.py

```diff
@@ -0,0 +1,681 @@
+        data_df.mad(axis=0)
+               .sort_values(ascending=False)
+    )
+    top_feats = mad_ranking[:subset_feats].index.astype(str).values
+    print(top_feats[:5])
+    data_df = data_df.reindex(top_feats, axis='columns')
+    
+print(data_df.shape)
+data_df.iloc[:5, :5]
+
+
+# ### Load Park et al. "hit" data
+# 
+# This was collated/formatted in `0_process_park.ipynb`
+
+# In[6]:
+
+
+with open(cfg.distance_gain_info, 'rb') as f:
+    park_gain_info = pkl.load(f)
+    
+park_gain_info['TP53_BRCA'].head()
+
+
+# In[7]:
+
+
+with open(cfg.distance_loss_info, 'rb') as f:
+    park_loss_info = pkl.load(f)
+    
+park_loss_info['TP53_BRCA'].head()
+
+
+# ### Calculate distance between means/medians for given gene + cancer type
+
+# In[8]:
+
+
+from scipy.spatial.distance import pdist, squareform
+
+def get_centroids_and_distance(identifier, info_df, centroid_method='mean'):
+    
+    groups = ['both', 'none', 'one']
+    group_combinations = list(it.combinations(groups, 2))
+    class_name = info_df['class_name'].values[0]
+    
+    # get expression data for samples
+    samples = info_df.index.intersection(data_df.index)
+    info_df = info_df.reindex(samples)
+    
+    # if one group has no samples, we have to make sure to assign it 0 count
+    class_counts = []
+    hit_class_counts = info_df.groupby('num_hits').count().class_name
+    for group in groups:
+        if group in hit_class_counts.index:
+            class_counts.append(hit_class_counts[group])
+        else:
+            class_counts.append(0)
+    
+    # group by number of hits, then calculate centroids
+    centroids_df = (data_df
+        .reindex(samples)
+        .merge(info_df['num_hits'], left_index=True, right_index=True)
+        .groupby('num_hits')
+    )
+    
+    if centroid_method == 'mean':
+        centroids_df = centroids_df.mean()
+    elif centroid_method == 'median':
+        centroids_df = centroids_df.median()
+    else:
+        raise NotImplementedError(
+            'centroid method {} not implemented'.format(centroid_method)
+        )
+    
+    # calculate distance between centroids
+    # make sure this is in the same order for each identifier, and
+    # handle NA distances here (if one group doesn't have any samples)
+    dists = pdist(centroids_df.values, metric='euclidean')
+    dist_combinations = list(it.combinations(hit_class_counts.index, 2))
+    ordered_dists = []
+    for cmb in group_combinations:
+        if cmb not in dist_combinations:
+            ordered_dists.append(np.nan)
+        else:
+            cmb_ix = dist_combinations.index(cmb)
+            ordered_dists.append(dists[cmb_ix])
+    
+    return (groups,
+            group_combinations,
+            class_counts,
+            ordered_dists)
+    
+get_centroids_and_distance('TP53_BRCA',
+                           park_loss_info['TP53_BRCA'],
+                           'median')
+
+
+# ### Calculate centroid distance between "hits", per class
```

Comment:
I'm getting really turned around on the terminology here. "Hits" means discovered to be recurrently mutated in the Park paper? And what does "class" mean here?

Reply:
And in scrolling up to write this comment, I answered my own question. Oops.